### PR TITLE
Workingbranch hs2158

### PR DESCRIPTION
**Pull request by inariDeeTee**
Implemented functionality to return list of commands to assist users.

### CODE REVIEW

File: newbank/server/NewBank.java [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
   private static final NewBank bank = new NewBank();
   private HashMap<String, Customer> customers;
+	private ArrayList<String> commands = new ArrayList<String>();
```
Comment (sjt-85) [inline newbank/server/NewBank.java]: tab

File: newbank/server/NewBank.java [hunk 2]
```diff
@@ -33,7 +34,14 @@ private void addTestData() {
     john.assignPassword("eccbc87e4b5ce2fe28308fd9f2a7baf3");
     customers.put("John", john);
   }
-
+  
+  private void addCommands(ArrayList<String> commands) {
```
Comment (sjt-85) [inline newbank/server/NewBank.java:40]: Hi Heather,
addCommands needs to be called in NewBank constructor.

File: newbank/server/NewBank.java [hunk 3]
```diff
@@ -65,6 +73,10 @@ public synchronized String processRequest(CustomerID customer, String request) {
           case "LOGOUT":
             customer.logOut();
             return logOut(customer);
+          case "COMMANDS" :
+              return listCommands(commands);
+          case "HELP" : 
+              return listCommands(commands);
```
Comment (sjt-85) [inline newbank/server/NewBank.java]: indentation = 4 spaces here

File: newbank/server/NewBank.java [hunk 4]
```diff
@@ -89,6 +101,16 @@ private String addNewAccount(CustomerID customerID, List<String> request) {
...
+  
+  private String listCommands(ArrayList<String> commands) {
+    String printCommands = new String();
+    for (String command : commands) {
+      printCommands += command;
+      printCommands += "\n";
+    }
+    printCommands = printCommands.substring(0, printCommands.length()-1);
+    return printCommands;
+	}
```
Comment (sjt-85) [inline newbank/server/NewBank.java]: tab

File: newbank/server/NewBank.java [hunk 5]
```diff
@@ -89,6 +101,16 @@ private String addNewAccount(CustomerID customerID, List<String> request) {
...
   }
+  
+  private String listCommands(ArrayList<String> commands) {
+    String printCommands = new String();
+    for (String command : commands) {
+      printCommands += command;
+      printCommands += "\n";
+    }
+    printCommands = printCommands.substring(0, printCommands.length()-1);
+    return printCommands;
```
Comment (sjt-85) [inline newbank/server/NewBank.java]: Could simplify to:
return printCommands.substring(0, printCommands.length()-1);

**Review by sjt-85 — CHANGES_REQUESTED** [review]
Hi Heather,
Looking really good.  Just a couple of changes needed then it's ready to merge.
Thanks

File: newbank/server/NewBank.java (hunk 1 above)
Comment (sjt-85) [reply newbank/server/NewBank.java]: also missing import statement for ArrayList

File: newbank/server/NewBank.java [hunk 6]
```diff
@@ -1,17 +1,20 @@
...
 public class NewBank {
 
   private static final NewBank bank = new NewBank();
   private HashMap<String, Customer> customers;
+  private ArrayList<String> commands = new ArrayList<String>();
 
   private NewBank() {
     customers = new HashMap<>();
     addTestData();
+    addCommands;
```
Comment (sjt-85) [inline newbank/server/NewBank.java]: needs to be addCommands(commands);

File: newbank/server/NewBank.java [hunk 7]
```diff
@@ -1,17 +1,20 @@
 package newbank.server;
 
 import java.util.Arrays;
+import java.util.ArrayList;
```
Comment (sjt-85) [inline newbank/server/NewBank.java:3]: ASCII sort order so upper case before lower case e.g.:
import java.util.ArrayList;
import java.util.Arrays;

File: newbank/server/NewBank.java (hunk 6 above)
Comment (inariDeeTee) [reply newbank/server/NewBank.java]: Lots of silly mistakes- you can tell I'm getting a bit frustrated with the process.

**Review by sjt-85 — APPROVED** [review]

**Pull request merged**